core/problems/problems.go

```diff
@@ -0,0 +1,81 @@
+package problems
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/otrego/clamshell/core/board"
+	"github.com/otrego/clamshell/core/color"
+	"github.com/otrego/clamshell/core/movetree"
+)
+
+// Flatten takes a Treepath and a MoveTree and returns
+// the root of the flat movetree. the flat movetree
+// ignores the last two nodes of the treepath
+func Flatten(tp movetree.Treepath, g *movetree.MoveTree) (*movetree.MoveTree, error) {
+	b, err := PopulateBoard(tp, g)
+	if err != nil {
+		return nil, err
+	}
+
+	gflat := movetree.New()
+	gflat.Root.Placements = b.GetFullBoardState()
+
+	aw := make([]string, 0)
+	ab := make([]string, 0)
+
+	for _, move := range gflat.Root.Placements {
```

Comment:
My work in https://github.com/otrego/clamshell/pull/161 should make this significantly simpler